docs/tutorial/tutorial.md

```diff
@@ -0,0 +1,129 @@
+
+```
+juju add-model wordpress-tutorial
+```
+
+### Deploy wordpress-k8s charm
+
+Deployment of WordPress requires a database integration. A preferred method in juju is through the
+native juju relation. In this case, a `mysql` interface is required by the wordpress-k8s charm and
+hence, [`mysql-k8s`](https://charmhub.io/mysql-k8s) charm will be used.
+
+Start off by deploying the wordpress charm. By default it will deploy the latest stable release of
+the wordpress-k8s charm.
+
+```
+juju deploy wordpress-k8s
+```
+
+If you wish to try out the latest features of the wordpress-k8s charm you can set an additional
+parameter to deploy from a specific channel. For example, to deploy from the edge channel:
+
+```
+juju deploy wordpress-k8s --channel=edge
+```
+
+### Deploy and relate database
+
+#### Database relation
+
+The following commands deploys the mysql-k8s charm and relates wordpress-k8s charm through its mysql
+interface.
+
+```
+juju deploy mysql-k8s \
+    --config mysql-interface-user=wordpress \
+    --config mysql-interface-database=test-database
+juju relate wordpress-k8s mysql-k8s
+```
+
+#### Database configuration
+
+Note: If you’ve completed the steps in the Database relation section, you may skip this part. If
+you want to deploy your wordpress-k8s charm through configuration, you may continue to follow the
+steps in this section.
+
+The commands below create the MySQL deployment in microk8s which we can then use to configure the
+database for wordpress-k8s. Note that the environment variables are for demonstration purposes only
+and should not be used for production environments.
+
+```
+microk8s kubectl run mysql -n wordpress-tutorial --image=mysql:latest \
+--env="MYSQL_ROOT_PASSWORD=mysecretpassword" \
+--env=”MYSQL_DATABASE=wordpress” \
+--env=”MYSQL_USER=wordpress” \
+--env=”MYSQL_PASSWORD=wordpress”
+WORDPRESS_IP=microk8s kubectl get pod mysql --template '{{.status.podIP}}'
+juju config wordpress-k8s \
+db_host=$WORDPRESS_IP \
+db_name=wordpress \
+db_user=wordpress \
+db_password=wordpress
+```
+
+Use the following command to watch your deployment progress through different stages of deployment.
+
+```
+watch -c juju status --color
+```
+
+### Get admin credentials
+
+After the database has been configured in step 2, you can now access the WordPress application by
+accessing the IP of a wordpress-k8s unit. To start managing WordPress as an administrator, you need
+to get the credentials for the admin account.
+
+By running the `get-initial-password` action on a wordpress-k8s unit, juju will read and fetch the
+admin credentials setup for you. You can use the following command below.
+
+```
+juju run-action wordpress-k8s/0 get-initial-password --wait
+```
+
+The result should look something similar to the contents below:
+
+```
+unit-wordpress-k8s-0:
+  UnitId: wordpress-k8s/0
+  id: "6"
+  results:
+    password: 7ioJgp2CS6iK7MrM9I-wCm48i25HaJLzeHT7MpEwdyk
```

Comment:
Should this be omitted e.g., `<password>`? Just so the reader doesn't think this is the actual password for them?

Reply:
How about `password: <password> # should look something like: 7ioJgp2CS6iK7MrM9I-wCm48i25HaJLzeHT7MpEwdyk` with a comment?
I think it would be helpful to show how the password might look like, since this is a pretty long password that isn't too common.

Reply:
Looks good, perhaps `password: <password> # should look something like: XXXXXXXXXXXXXXXXXX-XXXXXXXXXXXXXXXXXXXXXXXX`